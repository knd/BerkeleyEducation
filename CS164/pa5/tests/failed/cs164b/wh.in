def t = {};
t[0] = {};
t[0][0] = 'a';
t[0][1] = 'b';
t[1] = {};
t[1][0] = 'c';
t[1][1] = 'd';

def f(v, d) {
  if (d) {
    f(v[0], d - 1);
    f(v[1], d - 1);
  } else {
    yield(v);
  }
}

def g(function, arg1, arg2) {
  lambda(unused) {
    function(arg1, arg2);
  }
}

def i(c, p) {
  lambda() {
    resume(c, p);
  }
}

for (x in ['0x' + x + 0 for x in i(coroutine(g(f, t, 2)), null)]) {
  print x;
};
