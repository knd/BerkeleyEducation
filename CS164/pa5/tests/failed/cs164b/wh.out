[('def', 't', ('dict-lit', [])), ('put', ('var', 't'), ('int-lit', 0), ('dict-lit', [])), ('put', ('get', ('var', 't'), ('int-lit', 0)), ('int-lit', 0), ('string-lit', 'a')), ('put', ('get', ('var', 't'), ('int-lit', 0)), ('int-lit', 1), ('string-lit', 'b')), ('put', ('var', 't'), ('int-lit', 1), ('dict-lit', [])), ('put', ('get', ('var', 't'), ('int-lit', 1)), ('int-lit', 0), ('string-lit', 'c')), ('put', ('get', ('var', 't'), ('int-lit', 1)), ('int-lit', 1), ('string-lit', 'd')), ('def', 'f', ('lambda', ['v', 'd'], [('if', ('var', 'd'), [('exp', ('call', ('var', 'f'), [('get', ('var', 'v'), ('int-lit', 0)), ('-', ('var', 'd'), ('int-lit', 1))])), ('exp', ('call', ('var', 'f'), [('get', ('var', 'v'), ('int-lit', 1)), ('-', ('var', 'd'), ('int-lit', 1))]))], [('exp', ('yield', ('var', 'v')))])])), ('def', 'g', ('lambda', ['function', 'arg1', 'arg2'], [('exp', ('lambda', ['unused'], [('exp', ('call', ('var', 'function'), [('var', 'arg1'), ('var', 'arg2')]))]))])), ('def', 'i', ('lambda', ['c', 'p'], [('exp', ('lambda', [], [('exp', ('resume', ('var', 'c'), ('var', 'p')))]))])), ('for', 'x', ('comprehension', ('+', ('+', ('string-lit', '0x'), ('var', 'x')), ('int-lit', 0)), 'x', ('call', ('var', 'i'), [('coroutine', ('call', ('var', 'g'), [('var', 'f'), ('var', 't'), ('int-lit', 2)])), ('null',)])), [('print', ('var', 'x'))])]
